bin/create_hugo_data.py

```diff
@@ -494,13 +494,149 @@ def export_events(anthology, builddir, dryrun):
+                # Extract talk ID from video filename like "2022.acl-keynote.2.mp4"
+                video_name = talk.attachments["video"].name
+                if video_name.endswith(".mp4"):
+                    # Remove .mp4 extension to get the talk ID
+                    talk_id = video_name[:-4]
+                else:
+                    talk_id = video_name
+            else:
+                # Fallback to sequential numbering if no video
+                talk_id = f"{event.id}.talk-{idx}"
+
+            # Create talk data
+            talk_data = {
+                "id": talk_id,
+                "event_id": event.id,
+                "event_title": event.title.as_text() if event.title else f"{event.id}",
+                "title": talk.title.as_text(),
+                "title_html": remove_extra_whitespace(
+                    talk.title.as_html(allow_url=False)
+                ),
+                "speakers": [],
+                "type": talk.type,
+            }
+
+            # Process speakers
+            for speaker in talk.speakers:
+                speaker_data = {
+                    "id": speaker.id if speaker.id else "",
+                    "full_name": speaker.name.as_first_last(),
+                    "first": speaker.name.first if speaker.name.first else "",
+                    "last": speaker.name.last,
+                }
+                talk_data["speakers"].append(speaker_data)
+
+            # Add video URL if available
+            if "video" in talk.attachments:
+                talk_data["video_url"] = talk.attachments["video"].url
+                talk_data["video_name"] = talk.attachments["video"].name
+
+            # Add other attachments
+            attachments = []
+            for att_type, attachment in talk.attachments.items():
+                if att_type != "video":
+                    attachments.append(
+                        {
+                            "type": att_type,
+                            "url": attachment.url,
+                            "name": (
+                                attachment.name
+                                if hasattr(attachment, 'name')
+                                else att_type
+                            ),
+                        }
+                    )
+            if attachments:
+                talk_data["attachments"] = attachments
+
+            # Add location and dates from event
+            if event.location:
+                talk_data["location"] = event.location
+            if event.dates:
+                talk_data["dates"] = event.dates
+
+            all_talks[talk_id] = talk_data
+
+            # Write individual talk file
+            if not dryrun:
+                with open(f"{builddir}/data/talks/{talk_id}.json", "wb") as f:
+                    f.write(ENCODER.encode(talk_data))
+
+    # Note: Not creating combined talks.json to avoid Hugo data collisions
+    # Individual talk files in build/data/talks/ are sufficient
+
```

Comment:
The reason for only writing a single JSON file (for papers, volumes, etc.) is performance; we should really do the same here.

Reply:
You're right, fixed it now